### PR TITLE
Fixed fetching URL for Jenkins instance

### PR DESCRIPTION
**Pull request by sladyn98**

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ export function activate(context: vscode.ExtensionContext) {
     // The code you place here will be executed every time your command is executed
 
     // Display a message box to the user
-    const schemaURL = vscode.workspace.getConfiguration().get('jcasc.schemaURL', '')
+    var schemaURL = vscode.workspace.getConfiguration().get('jcasc.schemaURL', '')
```
Comment (jetersen) [inline src/extension.ts]: Than it should not be called `schemaURL` 🤔
```suggestion
    const jenkinsURL = vscode.workspace.getConfiguration().get('jcasc.jenkinsURL', '')
```

File: src/extension.ts [hunk 2]
```diff
@@ -51,6 +51,7 @@ export function activate(context: vscode.ExtensionContext) {
           Authorization: `Basic ${auth}`,
         },
       }
+      schemaURL = schemaURL + "/configuration-as-code/schema"
```
Comment (jetersen) [inline src/extension.ts]:
```suggestion
      const schemaURL = jenkinsURL + "/configuration-as-code/schema"
```

**Review by jetersen — CHANGES_REQUESTED** [review]

File: src/extension.ts (hunk 2 above)
Comment (jetersen) [reply src/extension.ts]: Also we should properly trim trailing slash or add trailing slash.
So you can either remove or keep the first `/` in `/configuration-as-code/schema`

File: src/extension.ts (hunk 1 above)
Comment (sladyn98) [reply src/extension.ts]: Yeah probably :+1:

File: src/extension.ts [hunk 3]
```diff
@@ -43,14 +43,15 @@ export function activate(context: vscode.ExtensionContext) {
...
-    if (schemaURL && userName && userToken) {
+    if (jenkinsURL && userName && userToken) {
       // Fetches the JSON Schema via a REST API Call
       const auth = Buffer.from(`${userName}:${userToken}`).toString('base64')
       const options: RequestInit = {
         headers: {
           Authorization: `Basic ${auth}`,
         },
       }
+      const schemaURL = jenkinsURL + "configuration-as-code/schema"
```
Comment (jetersen) [inline src/extension.ts]: I think we should have a method that than adds the trailing slash 😓

Comment (sladyn98) [reply src/extension.ts]: Sounds Cool to me :)

File: src/extension.ts [hunk 4]
```diff
@@ -51,7 +51,7 @@ export function activate(context: vscode.ExtensionContext) {
           Authorization: `Basic ${auth}`,
         },
       }
-      const schemaURL = jenkinsURL + "configuration-as-code/schema"
+      const schemaURL = jenkinsURL + addTrailingSlash("configuration-as-code/schema")
```
Comment (jetersen) [inline src/extension.ts]: it is not `configuration-as-code/schema` that needs the trailing slash.
It's jenkins URL and should only add it if does not end with a trailing slash.

Comment (sladyn98) [reply src/extension.ts]: In the Readme, I added that the user enter the Jenkins url in the format that it contains a trailing slash.Fine to adding it here though.

Comment (jetersen) [reply src/extension.ts]: not every user will do it, same reason why Jenkins Core has the same fix :)

**Review by jetersen — APPROVED** [review]
LGTM

**Review by timja — APPROVED** [review]

**Pull request merged**